src/main/scala/scoverage/ScoverageKeys.scala

```diff
@@ -11,7 +11,12 @@ object ScoverageKeys {
   lazy val coverageAggregate = taskKey[Unit]("aggregate reports from subprojects")
   lazy val coverageExcludedPackages = settingKey[String]("regex for excluded packages")
   lazy val coverageExcludedFiles = settingKey[String]("regex for excluded file paths")
-  lazy val coverageMinimum = settingKey[Double]("scoverage-minimum-coverage")
+  lazy val coverageMinimum = settingKey[Double]("scoverage-minimum-coverage-stmt-total")
```

Comment:
So digging into this, I've actually never realized that before this was just taking into account statements, and not actual branches. Some of the issues I've seen people report about confusing result not make way more sense since the percentages that are shown are not always logical without realizing this is just the stmt total.

However, this also make me think that we now have a way to set minimal statement coverage and minimal branch coverage. Together, those two will make the up the actual total number we are shooting for. I wonder if that will also cause a bit of confusion since there still may be an assumption that `coverageMinimum` _is_ the total. 🤔 

Part of me wants to actually repurpose `coverageMinimum` to actual make it a total value for averaging stmt and branch, because I do think a lot of people think that's what it means. Then actually move what this to a new `coverageMinimumStmtTotal` key. However, that's sort of breaking behavior, even if people wouldn't see how necessarily.

Just to be super clear, my worry is this. Given the following:

```scala
// build.sbt
coverageMinimum := 56,
coverageFailOnMinimum := true
```

And someone runs their tests and they see:

```
[info] Statement coverage.: 60.00%
[info] Branch coverage....: 50.00%
[info] Coverage reports completed
[info] All done. Coverage was [60.00%]
```

I think it may cause confusion. My mind would go towards, "how is my coverage at 60%"? I have 60% statement and 50% branch meaning a 55% total coverage. This should have failed.

What do you think?

Reply:
the main thing i wanted to do is preserve backwards compatibility, so if someone is not interested in these granular keys, their build still keeps working. is there a way to display a deprecation message which says, please use the "total statement" minimum instead of this setting?

Reply:
> the main thing i wanted to do is preserve backwards compatibility, so if someone is not interested in these granular keys, their build still keeps working.

Exactly this was my main point as well. Basically the change would force a 2.0 release. Let me think on this a bit. I think ultimately it's the direction that I want to go, but I think I'll want to warn people first. More than likely, if you're alright with it, I'll probably roll back that change on this pr, figure out a way to warn users that the meaning of the key will change in the future, so that there is at least some advance warning. Again, thanks for you thoughts and effort on this.

Reply:
no problem, whatever you prefer. again, since my understanding of sbt code should be written is rudimentary (i.e., simply parroted the existing code), i will leave it to the experts on how to handle this.

let me know if you want me to make another change. ultimately, i would like to close this out (and especially the maven version which is what we use in our environment).

Reply:
ok, found out how to deprecate a setting. removed the average computation. now it's backwards-compatible but with a warning.

Reply:
Alright, this is perfect. Thanks again for all your work on this and putting up with my back and forths.